File: Cxx/Boneyard/GeometricObjects/WriteFile/Quad.cxx
#include <vtkCellArray.h>
#include <vtkPoints.h>
#include <vtkQuad.h>
#include <vtkPolyData.h>
#include <vtkXMLPolyDataWriter.h>
#include <vtkSmartPointer.h>

int main(int, char *[])
{
  //Setup the coordinates of four points 
  //(must be in counter clockwise order)
  double P0[3] = {0.0, 0.0, 0.0};
  double P1[3] = {1.0, 0.0, 0.0};
  double P2[3] = {1.0, 1.0, 0.0};
  double P3[3] = {0.0, 1.0, 0.0};
      
  //Create the points
  vtkSmartPointer<vtkPoints> points = vtkSmartPointer<vtkPoints>::New();
  points->InsertNextPoint(P0);
  points->InsertNextPoint(P1);
  points->InsertNextPoint(P2);
  points->InsertNextPoint(P3);
  
  //Create a quad on the points
  vtkSmartPointer<vtkQuad> quad = vtkSmartPointer<vtkQuad>::New();
  quad->GetPointIds()->SetId(0,0);
  quad->GetPointIds()->SetId(1,1);
  quad->GetPointIds()->SetId(2,2);
  quad->GetPointIds()->SetId(3,3);
  
  //Add the quad to a cell array
  vtkSmartPointer<vtkCellArray> quads = vtkSmartPointer<vtkCellArray>::New();
  quads->InsertNextCell(quad);

  //Add the points and quad to a polydata
  vtkSmartPointer<vtkPolyData> polydata = vtkSmartPointer<vtkPolyData>::New();
  polydata->SetPoints(points);
  polydata->SetPolys(quads);

  //Write the file
  vtkSmartPointer<vtkXMLPolyDataWriter> writer = vtkSmartPointer<vtkXMLPolyDataWriter>::New();
  writer->SetInput(polydata);
  writer->SetFileName("Quad.vtp");
  writer->Write();

  return EXIT_SUCCESS;
}

File: src/CSharp/GeometricObjects/Quad.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Windows.Forms;
using System.Diagnostics;

using Kitware.VTK;

namespace ActiViz.Examples {
   public partial class Form1 : Form {
      public Form1() {
         InitializeComponent();
      }


      private void renderWindowControl1_Load(object sender, EventArgs e) {
         try {
            Quad();
         }
         catch(Exception ex) {
            MessageBox.Show(ex.Message, "Exception", MessageBoxButtons.OK);
         }
      }


      private void Quad() {
         double[,] p = new double[,] { 
            { 0.0, 0.0, 0.0 }, 
            { 1.0, 0.0, 0.0 },
            { 1.0, 1.0, 0.0 },
            { 0.0, 1.0, 0.0 }
         };

         // Create the points
         vtkPoints points = vtkPoints.New();
         for(int i = 0; i < 4; i++)
            points.InsertNextPoint(p[i, 0], p[i, 1], p[i, 2]);


         vtkQuad quad = vtkQuad.New();
         quad.GetPointIds().SetNumberOfIds(4);
         for(int i = 0; i < 4; i++)
            quad.GetPointIds().SetId(i, i);

         // Create a cell array to store the quad in and add the quad to it
         vtkCellArray cells = vtkCellArray.New();
         cells.InsertNextCell(quad);

         // Create a polydata to store everything in
         vtkPolyData polyData = vtkPolyData.New();

         // Add the points to the dataset
         polyData.SetPoints(points);

         // Add the quad to the dataset
         polyData.SetPolys(cells);

         //Create an actor and mapper
         vtkPolyDataMapper mapper = vtkPolyDataMapper.New();
         mapper.SetInput(polyData);
         vtkActor actor = vtkActor.New();
         actor.SetMapper(mapper);
         vtkRenderWindow renderWindow = renderWindowControl1.RenderWindow;
         vtkRenderer renderer = renderWindow.GetRenderers().GetFirstRenderer();
         renderer.SetBackground(0.2, 0.3, 0.4);
         renderer.AddActor(actor);
      }
   }
}
